spoc/io.py

```diff
@@ -17,16 +20,21 @@
 )
 from spoc.fragments import Fragments
 
+# Instantiate one duckdb connection to be used for all duckdb relations
+DUCKDB_CONNECTION = duckdb.connect(database=":memory:")
+
 
 class FileManager:
     """Is responsible for loading and writing files
 
     Args:
-        use_dask (bool, optional): Whether to use Dask for reading Parquet files. Defaults to False.
+        data_mode (DataMode, optional): Data mode. Defaults to DataMode.PANDAS.
     """
 
-    def __init__(self, use_dask: bool = False) -> None:
-        if use_dask:
+    def __init__(self, data_mode: DataMode = DataMode.PANDAS) -> None:
+        if data_mode == DataMode.DUCKDB:
+            self._parquet_reader_func = partial(duckdb.read_parquet, connection=DUCKDB_CONNECTION)
+        elif data_mode == DataMode.DASK:
             self._parquet_reader_func = dd.read_parquet
         else:
```

Comment:
Could it be that a user would submit an incorrect data mode? This clause won't catch that and reset to the default pandas backend.

Alternative would be `elif data_mode == DataMode.PANDAS: ... else: raise ValueError`.

Reply:
It might sound too pedantic, but it would be expressive to implement such structure of branching when evaluating the data mode everywhere... Although, for us it shouldn't matter much since we know that we have only 3 data modes, but for other readers of the code or us in the future it would be a nice explicit explanation.

Btw, python 3.10 supports `switch ... case`, but that would require upping the python version.

Reply:
Fair, I added the check for supported datamodes!